File: web/src/lib/sync/path-encoder.ts
import path from "path";
import os from "os";

export function encodeProjectPath(absPath: string): string {
  return absPath.replace(/[/.]/g, "-");
}

export function extractDisplayName(projectPath: string): string {
  return path.basename(projectPath) || projectPath;
}

export function getClaudeDir(): string {
  return process.env.CLAUDE_DIR || path.join(os.homedir(), ".claude");
}

export function getProjectsDir(): string {
  return path.join(getClaudeDir(), "projects");
}

File: src-tauri/src/sync/path_encoder.rs
use std::path::{Path, PathBuf};

pub fn get_claude_dir() -> PathBuf {
    if let Ok(dir) = std::env::var("CLAUDE_DIR") {
        return PathBuf::from(dir);
    }
    dirs::home_dir()
        .expect("Could not determine home directory")
        .join(".claude")
}

pub fn get_projects_dir() -> PathBuf {
    get_claude_dir().join("projects")
}

pub fn extract_display_name(project_path: &str) -> String {
    Path::new(project_path)
        .file_name()
        .map(|s| s.to_string_lossy().to_string())
        .unwrap_or_else(|| project_path.to_string())
}
